### DISCUSSION TITLE
Monthly Wallet and Dapp Meeting #24 [June] 💬

### DISCUSSION THREAD

**Discussion by arthurgousset** [celo-org/celo-monorepo#10374] [Wallet and (d)App Council]
# Monthly Wallet and Dapp Meeting

##  📆 Quick links 

- When: **Wednesday, June 14 (9am PT/6pm CET)** 
- Where: **[Google Meet](https://meet.google.com/uwj-wfse-uzm)**
- Agenda: **comments**👇
- Previous: https://github.com/celo-org/celo-monorepo/discussions/10308
- Next: https://github.com/celo-org/celo-monorepo/discussions/10421
- Recording: after meeting

## 🤷‍♂️ What is this?

A monthly meeting for Celo **developers, founders,** **product** people (anyone really!) to come together and
- find consensus (coordinate),
- learn about needs, pain points, and desires
- present and collect technical feedback

## 🗒️  Agenda

- Comment with a short sentence describing what you'd like to discuss
- Don't worry, it can be small or big! 
- Every builder is welcome (and we are very easy going, I promise 😄)
- Upvote topics you're interested in

## Housekeeping

_You can safely ignore this!_

- [x] Send update calendar invite with Github link
- [ ] Encourage additions to the agenda
- [ ] Publicise on forum and Discord
- [ ] RECORD meeting 🎥 
- [ ] Share recording below and on forum
- [ ] Create next meeting (link to next meeting above)

**Comment by arthurgousset** [comment]
## WalletConnect v2 upgrades (planned WalletConnect v1 shutdown)

### Reminder 

+ On **June 28, 2023**, WalletConnect v1 bridge servers will be shutdown. 
+ If (i) you are a **dapp developer,** (ii) WalletConnect v1 bridge servers are shutdown, and (iii) you don't upgrade your dapp to WalletConnect v2, users will not be able to connect to your dapp from 3rd party wallets via WalletConnect.
+ If (i) you are a **wallet developer**, ii) WalletConnect v1 bridge servers are shutdown,  and (iii) you don't support WalletConnect v2, users will not be able to connect to 3rd party dapps from your wallet via WalletConnect.

### Dapp developers using [`rainbowkit`](https://www.npmjs.com/package/@rainbow-me/rainbowkit)

TLDR: 👉 **You must be on >v0.12._x_.**
+ [NPM releases](https://www.npmjs.com/package/@rainbow-me/rainbowkit?activeTab=versions)
+ [Github release notes](https://github.com/rainbow-me/rainbowkit/releases)

From the [Rainbowkit migration guide](https://www.rainbowkit.com/docs/migration-guide#012x-breaking-changes):

  > RainbowKit has adopted the `WalletConnectLegacyConnector` connector in `wagmi` for continued WalletConnect v1 support.
  > Support for WalletConnect v2 and `WalletConnectConnector` will soon be available as a patch release, without breaking changes.
  > 
  > Wallets will be transitioned automatically in future releases.
  > 
  > Every dApp must now provide a [WalletConnect Cloud](https://cloud.walletconnect.com/) `projectId` to enable WalletConnect v2. This must be completed before WalletConnect v1 bridge servers are shutdown on **June 28, 2023**. RainbowKit will quietly prefer v1 for all wallets if `projectId` is unspecified.
  > Follow the [steps below](https://www.rainbowkit.com/docs/migration-guide#012x-breaking-changes) to migrate.

### Dapp developers using [`react-celo`](https://www.npmjs.com/package/@celo/react-celo) 

TLDR: 👉 **You must be on >v5.0.4**
+ [NPM release v5.0.4](https://www.npmjs.com/package/@celo/react-celo/v/5.0.4)
+ [Github release notes v5.0.4](https://github.com/celo-org/react-celo/releases/tag/v5.0.4)

## Team check-in (if folks are present)

  - Valora WCv2/WCv1 approach, I believe Valora supports both 🥳 
  - ChurritoFi upgrade (Nico)
  - Ubeswap upgrade/help (Dan)
  - NodeFi

**Reply by arthurgousset** [reply]
## Meeting notes: 

- Valora supports WCv2 and WCv1. Valora was contacted by WalletConnect team directly.
  - Nuance about Sign and Auth clients in the new WalletConnect library. WC team considers WCv2 support met if wallets implement both the Sign and Auth client.
  - This nuance shouldn't prevent dapps to connect via WCV2.
  - Uniswap and Pancake swap have a staging deployment for WCv2 support.
  - Jean will check if he can share the staging URLs for Uniswap and Pancake Swap
  - None of the dapps that are listed in Valora have not upgrade to WCv2
- ImpactMarket (has both wallet and dapp perspective)
  - rainbowkit didn't have WCv2 working because they were blocked by wagmi.
  - Bernardo was waiting on rainbowkit for a while and simply decided to us web3modal (by WalletConnect team) instead
  - web3wallet (by WalletConnect team) is a very simple library to support both WCv1 and WCv2 ("just a few lines of code for session management"). web3modal integrates with wagmi.
  - Ledger had some problems with WCv2
  - wagmi was blocker on WCv2 and deprecated ethers support 
- Nestor question about recommendations
  - on the wallet side definitely web3wallet
  - on the dapp side rainbowkit or web3modal
- Ubeswap
  - a few weeks ago Silas made a PR to upgrade to react-celo
  - https://github.com/Ubeswap/ubeswap-interface/pull/676
- NodeFinance (Libera)
  - No dependency on WalletConnect.
  - But Libera has WCv2 support

**Comment by arthurgousset** [comment]
## Use of Blockscout’s GraphQL endpoint

> What problems does the Blockscout GraphQL endpoint solve for you?

I'd be interested to learn more about Valora's use of the GraphQL API (at `explorer.celo.org/mainnet/graphql`).
Specifically:

+ [ ] What type of data is served to users (how that relates to queries)
+ [ ] What alternative GraphQL APIs exist on Celo (if they know of any)
+ [ ] Whether REST APIs that support internal transaction traces (be in it [Blockscout](https://explorer.celo.org/mainnet/api-docs) or others) are sufficient to satisfy Valora's user need (re Celo native transfers)

**Comment by arthurgousset** [comment]
## Use of QuickNode's Webhook API

> What problems do webhook APIs solve for you?

QuickNode recently release a product called “QuickAlerts” ([dashboard.quicknode.com/quick-alerts](http://dashboard.quicknode.com/quick-alerts)) that allows you to set up webhook alerts for on-chain events from arbitrary smart contracts on Celo. 

See: [Introducing QuickAlerts: Real-Time Alerts for On-Chain Events](https://blog.quicknode.com/introducing-quickalerts-real-time-alerts-for-on-chain-events/) [Feb 2023].

<img src="https://github.com/celo-org/celo-monorepo/assets/46296830/bb5a10e5-083e-4113-922b-16372463194c" width="800">

<img src="https://github.com/celo-org/celo-monorepo/assets/46296830/2751c727-4852-4e92-b0e1-d50c92c6b9a5" width="800">

<img src="https://github.com/celo-org/celo-monorepo/assets/46296830/87bd4e38-7f22-44b0-a39d-32547cf17628" width="800">

<img src="https://github.com/celo-org/celo-monorepo/assets/46296830/c4b684ee-4e4a-48d5-b283-bcb3ecd9d6f5" width="800">

<img src="https://github.com/celo-org/celo-monorepo/assets/46296830/1367af37-e03e-42ad-9a26-d43f2f9dcfd9" width="800">

**Comment by arthurgousset** [comment]
## (FYI) SocialConnect demo (for visibility)

The main demos are:
- This [cross-wallet payment video](https://www.loom.com/share/8afddd73ba324ec18aeb63fc96d568f9) (1minute)
- This [live demo website](https://www.socialconnect.nodefinance.org/) where you can register and lookup addresses using phone numbers (hosted by Node Finance)
- This 3 part video explainer on YouTube by DevRel
  - [Celo Spark: SocialConnect Mini-Series (1/3) — What Is It?](https://www.youtube.com/watch?v=a_756GRPcV4&list=PLsQbsop73cfHn2VjMMrrbdBjZFPYEgBp6&index=7)
  - [Celo Spark: SocialConnect Mini-Series (2/3) — How Does It Work?](https://www.youtube.com/watch?v=bzZbfoPLYM4&list=PLsQbsop73cfHn2VjMMrrbdBjZFPYEgBp6&index=8)
  - [Celo Spark: SocialConnect Mini-Series (3/3) — Coding Session](https://www.youtube.com/watch?v=qrIHC496avs&list=PLsQbsop73cfHn2VjMMrrbdBjZFPYEgBp6&index=9)
- This [lookup demo using twitter handles](https://github.com/celo-org/SocialConnect-Twitter) by DevRel

**Comment by jeanregisser** [comment]
## Same device deeplinking (wallet <--> dapp connections)

Improving UX to connect to dapps (in-app browser, or OS browser) when started from a wallet.

**Reply by arthurgousset** [reply]
Meeting notes: 

- Jean - Connecting to dapps (on the **same device**) from Valora requires a lot of steps ("connect", "WalletConnect" or "Valora", ...)
  - on iOS multiple steps
  - on Android a step less
  - Idea: make proposal in rainbowkit or web3modal that improves in-app browser experience (by recognising the wallet the in-app browser is in)
- Frederic - UX is particularly difficult when going back and forth between mobile device and dapp.
  - any tests to go back from dapp to wallet?
  - Jean - yes in the past used deeplinks in dappkit (which was sunset because features were limited).
  - https://github.com/celo-org/celo-monorepo/discussions/10221
  - Frederic - deeplinks is something is often missed by mobile wallets
- ImpactMarket - has also thought about easier ways to connect in-app (rather than needing to connect, choosing wallet, etc) 
- Jean - technical spec would specify some wallet field and dapp field.
- Frederic - OpenID is thinking about this a lot

**Comment by jeanregisser** [comment]
WC v1 shutdown deadline, who has upgraded?

**Reply by arthurgousset** [reply]
(for future reference)
We touched on this in the [discussion topic at the top](https://github.com/celo-org/celo-monorepo/discussions/10374#discussioncomment-6174704)
